File: scripts/convert-poses-to-grayscale.py
#!/usr/bin/env python3

"""
Convert pose reference images to grayscale
This prevents AI from copying clothing colors from pose references
"""

from PIL import Image
import os

# Configuration
POSE_NUMBERS = [1, 2, 3, 4, 5]
SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
INPUT_DIR = os.path.join(SCRIPT_DIR, '../assets/poses')
OUTPUT_DIR = os.path.join(SCRIPT_DIR, '../assets/poses/grayscale')

def convert_to_grayscale(input_path, output_path):
    """Convert a single image to grayscale"""
    try:
        # Open the image
        img = Image.open(input_path)
        
        # Convert to grayscale (keep alpha channel if present)
        if img.mode == 'RGBA':
            # Separate RGB and alpha
            rgb = img.convert('RGB')
            gray = rgb.convert('L')
            # Merge back with alpha
            img_gray = Image.merge('LA', (gray, img.split()[3]))
            # Convert back to RGBA for PNG
            img_gray = img_gray.convert('RGBA')
        else:
            img_gray = img.convert('L').convert('RGBA')
        
        # Save as PNG
        img_gray.save(output_path, 'PNG')
        print(f"✓ Converted: {os.path.basename(output_path)}")
        return True
    except Exception as e:
        print(f"✗ Failed to convert {os.path.basename(input_path)}: {e}")
        return False

def main():
    # Create output directory
    os.makedirs(OUTPUT_DIR, exist_ok=True)
    
    print("Converting pose reference images to grayscale...\n")
    
    converted = 0
    for pose_num in POSE_NUMBERS:
        padded_num = str(pose_num).zfill(2)
        input_path = os.path.join(INPUT_DIR, f"pose{padded_num}.png")
        output_path = os.path.join(OUTPUT_DIR, f"pose{padded_num}.png")
        
        if os.path.exists(input_path):
            if convert_to_grayscale(input_path, output_path):
                converted += 1
        else:
            print(f"⚠ Skipped: pose{padded_num}.png (file not found)")
    
    print(f"\n✓ Done! Converted {converted} images to grayscale")
    print(f"   Saved to: {OUTPUT_DIR}")
    print("\nNext steps:")
    print("1. Upload these grayscale images to R2 storage at:")
    print("   little-hero-assets/book-mvp-simple-adventure/characters/poses/")
    print("2. In n8n workflow, delete 'Convert Poses to Grayscale' node")
    print("3. Connect 'Load Pose Reference' directly to 'Prepare Gemini Requests'")

if __name__ == "__main__":
    main()

File: scripts/convert-poses-to-grayscale.js
#!/usr/bin/env node

/**
 * Convert pose reference images to grayscale
 * This prevents AI from copying clothing colors from pose references
 */

const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

// Configuration
const POSE_NUMBERS = [1, 2, 3, 4, 5];
const INPUT_DIR = path.join(__dirname, '../assets/poses');
const OUTPUT_DIR = path.join(__dirname, '../assets/poses/grayscale');

// Create output directory
if (!fs.existsSync(OUTPUT_DIR)) {
  fs.mkdirSync(OUTPUT_DIR, { recursive: true });
}

async function convertToGrayscale(inputPath, outputPath) {
  try {
    await sharp(inputPath)
      .grayscale()
      .toFile(outputPath);
    console.log(`✓ Converted: ${path.basename(outputPath)}`);
  } catch (error) {
    console.error(`✗ Failed to convert ${path.basename(inputPath)}:`, error.message);
  }
}

async function main() {
  console.log('Converting pose reference images to grayscale...\n');

  for (const poseNum of POSE_NUMBERS) {
    const paddedNum = poseNum.toString().padStart(2, '0');
    const inputPath = path.join(INPUT_DIR, `pose${paddedNum}.png`);
    const outputPath = path.join(OUTPUT_DIR, `pose${paddedNum}.png`);

    if (fs.existsSync(inputPath)) {
      await convertToGrayscale(inputPath, outputPath);
    } else {
      console.log(`⚠ Skipped: pose${paddedNum}.png (file not found)`);
    }
  }

  console.log('\n✓ Done! Grayscale images saved to:', OUTPUT_DIR);
  console.log('\nNext steps:');
  console.log('1. Upload these grayscale images to R2 storage at:');
  console.log('   little-hero-assets/book-mvp-simple-adventure/characters/poses/');
  console.log('2. Remove "Convert Poses to Grayscale" node from workflow');
  console.log('3. Connect "Load Pose Reference" directly to "Prepare Gemini Requests"');
}

main().catch(console.error);




